docs/source/api_reference/forecasting.rst

```diff
@@ -363,6 +363,16 @@ Transformer (deep learning) based forecasters
     LTSFDLinearForecaster
     LTSFNLinearForecaster
 
+RNN (deep learning) based forecasters
+-------------------------------------
+
+.. currentmodule:: sktime.forecasting.neuralforecast
+
+.. autosummary::
+    :toctree: auto_generated/
+    :template: class.rst
+
+    NeuralForecastRNN
```

Comment:
@benHeid Please let me know if you have suggestion for this grouping.

(and of course the general review of the PR, especially your opinion on pyproject.toml final state.)